test/lib/test-ringpop-cluster.js

```diff
@@ -127,12 +126,66 @@ function testRingpopCluster(opts, name, test) {
     }
 
     tape(name, function onTest(assert) {
-        var cluster = bootstrapClusterOf(opts, function onBootstrap(results) {
-            assert.on('end', function onEnd() {
-                destroyCluster(cluster);
+        var tmpMinProtocolPeriods = [];
+        var cluster = createClusterOf(opts);
+        var joinResults;
+
+        // store protocol period. Set protocol period to 1 so that convergence is fast
+        cluster.forEach(function each(ringpop, i) {
+            tmpMinProtocolPeriods[i] = ringpop.gossip.minProtocolPeriod;
+            ringpop.gossip.minProtocolPeriod = 1;
+        });
+
+        function onConverged() {
+            // do not run onConverged if not all joins are succesful
+            if (joinResults === undefined) {
+                return;
+            }
+
+            // revert back to original protocol period
+            cluster.forEach(function each(ringpop, i) {
+                ringpop.gossip.minProtocolPeriod = tmpMinProtocolPeriods[i];
             });
 
-            test(results, cluster, assert);
+            if (typeof opts.tapAfterConvergence === 'function') {
+                opts.tapAfterConvergence(cluster);
+            }
+
+            if (opts.waitForConvergence !== false && opts.checkChecksums === true) {
```

Comment:
this is getting a bit messy. can you extract these 10-12 lines of code into a function like:

```
function assertEqualChecksums(...) {
  ...
}
```